### PR TITLE
SISRP-13318 Treat retracted acceptance as unaccepted applicant

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-13318

### CODE REVIEW

File: spec/models/berkeley/user_roles_spec.rb [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    end
+
+    context 'no affiliations' do
+      let(:affiliations) { [] }
+      it 'should return no attributes' do
+        expect(subject[:roles]).to eq({})
+        expect(subject[:ug_grad_flag]).to be_nil
+      end
+    end
+  end
```
Comment (christianvuerings) [inline spec/models/berkeley/user_roles_spec.rb]: @raydavis thanks for these extra tests!

**Comment by christianvuerings** [comment]
lgtm - but might be good for another back-end dev to have a look at this.

**Comment by paulkerschen** [comment]
@raydavis I applaud moving the role transformation into Berkeley::UserRoles. Now that all the transformation tests have been moved into `spec/models/berkeley/user_roles_spec.rb`, should they be deleted from `spec/models/hub_edps/user_attributes_spec.rb`, or is there a case for leaving them in as duplicated?

**Comment by raydavis** [comment]
I wanted to leave them in temporarily, @paulkerschen , but yeah, my next step would usually be to clear the duplication out. I'll do that now & amend the PR.

**Comment by raydavis** [comment]
PR amended

**Comment by paulkerschen** [comment]
Thanks @raydavis!

**Pull request merged**